### PR TITLE
Added virtualenv for dependency management

### LINKED ISSUE #85: Add virtualenv for dependency management
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [X] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/badgeyay

Right now the website has crashed due to importerror of urllib module. What i have proposed is to avoid this issue in future i have created a virtual environment so that whenever someone uses a new module, they can `pip freeze > requirements.txt` it for dependency management.

@niccokunzmann @jajodiaraghav @gabru-md Please give your opinion

### PR DESCRIPTION
**Pull request by S2606**
Fixes #85 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:

-
-
-

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
 /*.csv#
 *.badges
 *.swp
-/*.pyc
\ No newline at end of file
+/*.pyc
```
Comment (niccokunzmann) [inline .gitignore]: `/*.pyc` ignores all pyc files in the root directory. `*.pyc` ignores all .pyc files in all dorectories. 
Github Pyhton ignore file: https://github.com/github/gitignore/blob/master/Python.gitignore

File: app/generate_csv_eventyay.py [hunk 2]
```diff
@@ -1,7 +1,12 @@
 #!usr/bin/python3
-from urllib2 import urlopen
 import json
 import os
+try:
+    # For Python 3.0 and later
+    from urllib.request import urlopen
+except ImportError:
+    # Fall back to Python 2's urllib2
+    from urllib2 import urlopen
```
Comment (niccokunzmann) [inline app/generate_csv_eventyay.py]: Best would be to put this in a separate pull-requets. Thus, if there are any discussions here about anything else,

- this gets merged asap
- other people can build on it

File: app/requirements.txt [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
-markupsafe==1.0
-werkzeug==0.12.2
-wtforms==2.1
+Jinja2==2.9.6
+MarkupSafe==1.0
 pyflakes==1.6.0
-gunicorn
\ No newline at end of file
+requests==2.18.4
+urllib3==1.22
```
Comment (niccokunzmann) [inline app/requirements.txt]: adding urllib and requests - we can one of those

File: .gitignore (hunk 1 above)
Comment (niccokunzmann) [reply .gitignore]: Please remove this file: https://github.com/fossasia/badgeyay/pull/86/files#diff-60cfe20702be862f7a97b99fea0b072a

Comment (S2606) [reply .gitignore]: @niccokunzmann Sorry I didnt get it

File: app/generate_csv_eventyay.py (hunk 2 above)
Comment (S2606) [reply app/generate_csv_eventyay.py]: @niccokunzmann ok

File: app/requirements.txt (hunk 3 above)
Comment (S2606) [reply app/requirements.txt]: without requesr urllib wont work

**Comment by jajodiaraghav** [comment]
@S2606 Why did you create a new commit? I told you to use `git commit --amend --no-edit`

**Comment by S2606** [comment]
@jajodiaraghav Sorry for my mistake. I had a doubt if its ok to resolve conflict using github rather than git?

**Pull request closed without merging**